src/app.ts

```diff
@@ -7,79 +7,98 @@ import { importText, FSHDocument, FSHTank } from './import';
 import { exportFHIR } from './export';
 import { IGExporter } from './ig/IGExporter';
 import { logger, stats } from './utils/FSHLogger';
+import { loadDependency } from './fhirdefs/load';
+import { FHIRDefinitions } from './fhirdefs';
 
-let input: string;
+app();
 
-program
-  .name('sushi')
-  .usage('<path-to-fsh-defs> [options]')
-  .option('-o, --out <out>', 'the path to the output folder', path.join('.', 'build'))
-  .arguments('<path-to-fsh-defs>')
-  .action(function(pathToFshDefs) {
-    input = pathToFshDefs;
-  })
-  .parse(process.argv);
+async function app() {
+  let input: string;
 
-// Check that input folder is specified
-if (!input) {
-  logger.error('Missing path to FSH definition folder.');
-  program.help();
-}
+  program
+    .name('sushi')
+    .usage('<path-to-fsh-defs> [options]')
+    .option('-o, --out <out>', 'the path to the output folder', path.join('.', 'build'))
+    .arguments('<path-to-fsh-defs>')
+    .action(function(pathToFshDefs) {
+      input = pathToFshDefs;
+    })
+    .parse(process.argv);
 
-let files: string[];
-try {
-  files = fs.readdirSync(input, 'utf8');
-} catch {
-  logger.error('Invalid path to FSH definition folder.');
-  program.help();
-}
+  // Check that input folder is specified
+  if (!input) {
+    logger.error('Missing path to FSH definition folder.');
+    program.help();
+  }
 
-const docs: FSHDocument[] = [];
-for (const file of files) {
-  if (file.endsWith('.fsh')) {
-    const filePath = path.resolve(input, file);
-    const fileContent = fs.readFileSync(filePath, 'utf8');
-    const doc = importText(fileContent, filePath);
-    if (doc) docs.push(doc);
+  let files: string[];
+  try {
+    files = fs.readdirSync(input, 'utf8');
+  } catch {
+    logger.error('Invalid path to FSH definition folder.');
+    program.help();
   }
-}
 
-let config: any;
-try {
-  config = JSON.parse(fs.readFileSync(path.join(input, 'package.json'), 'utf8').toString());
-} catch {
-  logger.error('No package.json in FSH definition folder.');
-  program.help();
-}
+  let config: any;
+  try {
+    config = JSON.parse(fs.readFileSync(path.join(input, 'package.json'), 'utf8').toString());
+  } catch {
+    logger.error('No package.json in FSH definition folder.');
+    program.help();
+  }
 
-const tank = new FSHTank(docs, config);
-const outPackage = exportFHIR(tank);
+  // Load external dependencies
+  const defs = new FHIRDefinitions();
+  const dependencyDefs: Promise<FHIRDefinitions>[] = [];
+  for (const dep of Object.keys(config?.dependencies ?? {})) {
+    try {
+      dependencyDefs.push(loadDependency(dep, config.dependencies[dep], defs));
+    } catch (e) {
+      logger.error(e.message);
+      logger.error(`Failed to load ${dep}#${config.dependencies[dep]}`);
+    }
```

Comment:
A normal try-catch block doesn't work here because `loadDependency` is an async function. When a `throw` statement runs in an async function, the function returns a rejected promise. So, you'll need to handle that with something like:
`loadDependency(...).catch(e => { /* current catch block goes here */ });`